src/main/java/duke/FileUpdater.java

```diff
@@ -0,0 +1,120 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+
+
+public class FileUpdater {
+    private File f;
+    private TaskList lst;
+
+    /**
+     * Constructor for a fileupdater class
+     * @param f, file to be read by Duke
+     * @param lst, tasklist for the list of tasks to be modified
+     * @throws FileNotFoundException
+     */
+    public FileUpdater(File f, TaskList lst) throws FileNotFoundException {
+        this.f = f;
+        this.lst = lst;
+
+        if (!f.exists()) {
+            throw new FileNotFoundException("file does not exist!");
+        }
+
+    }
+
+    /**
+     * method to return a file inputted into fileupdater
+     *
+     */
+    public File getFile(){
+        return this.f;
+    }
+
+    /**
+     * method to rewrite a file to sync the file with the updated tasklist
+     */
+    public void updateListFile(){
+        String s ="";
+        for(int i = 0; i < lst.size(); i++){
```

Comment:
Perhaps more spaces could be added to adhere to the coding standard? (Namely, before `""` and after `for`.)

Reply:
There are other instances in your code where it could be better if a space is left as well, maybe another check can be performed?